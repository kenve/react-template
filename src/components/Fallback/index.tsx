import React from 'react';
import { Spin } from 'antd';

const Fallback = (
  <div style={{ height: '100%', textAlign: 'center', opacity: '0.5' }}>
    <Spin data-testid="fallback-spinner" style={{ top: 'calc(50% - 18px)' }} />
  </div>
);

export default Fallback;
