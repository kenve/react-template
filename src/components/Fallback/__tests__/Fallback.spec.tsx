import React from 'react';
import { render } from '@testing-library/react';
import Fallback from '..';

describe('React element that is rendered while waiting for the component to load', () => {
  it('should render fallback correctly', () => {
    const { asFragment } = render(<>{Fallback}</>);
    expect(asFragment()).toMatchSnapshot();
  });
});
