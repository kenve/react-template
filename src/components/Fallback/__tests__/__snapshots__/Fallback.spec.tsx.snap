// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React element that is rendered while waiting for the component to load should render fallback correctly 1`] = `
<DocumentFragment>
  <div
    style="height: 100%; text-align: center; opacity: 0.5;"
  >
    <div
      class="ant-spin ant-spin-spinning"
      data-testid="fallback-spinner"
      style="top: calc(50% - 18px);"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
